src/Ng/Jobs/BackupV3Storage.cs

```diff
@@ -0,0 +1,236 @@
+                   + $"[-{Arguments.ValidateCertificate} true|false]]";
+        }
+
+        protected override void Init(IDictionary<string, string> arguments, CancellationToken cancellationToken)
+        {
+            _luceneStorageAccountName = arguments.GetOrThrow<string>(Arguments.LuceneStorageAccountName);
+            _luceneStorageKeyValue = arguments.GetOrThrow<string>(Arguments.LuceneStorageKeyValue);
+            _luceneStorageContainer = arguments.GetOrThrow<string>(Arguments.LuceneStorageContainer);
+
+            _catalogStorageAccountName = arguments.GetOrThrow<string>(Arguments.CatalogStorageAccountName);
+            _catalogStorageKeyValue = arguments.GetOrThrow<string>(Arguments.CatalogStorageKeyValue);
+            _catalogStorageContainer = arguments.GetOrThrow<string>(Arguments.CatalogStorageContainer);
+
+            _registrationStorageAccountName = arguments.GetOrThrow<string>(Arguments.RegistrationStorageAccountName);
+            _registrationStorageKeyValue = arguments.GetOrThrow<string>(Arguments.RegistrationStorageKeyValue);
+            _registrationStorageContainer = arguments.GetOrThrow<string>(Arguments.RegistrationStorageContainer);
+            _registrationgzStorageContainer = arguments.GetOrThrow<string>(Arguments.RegistrationGZStorageContainer);
+
+            _destStorageAccountName = arguments.GetOrThrow<string>(Arguments.DestStorageAccountName);
+            _destStorageKeyValue = arguments.GetOrThrow<string>(Arguments.DestStorageKeyValue);
+
+            _backupDays = arguments.GetOrDefault<int>(Arguments.BackupDays, DefaultBackupDays);
+        }
+
+        protected override async Task RunInternal(CancellationToken cancellationToken)
+        {
+            List<Task> deleteTasks = new List<Task>();
+            DateTime currentDate = DateTime.UtcNow;
+
+            deleteTasks.Add(TryDeleteContainerAsync(_luceneStorageContainer, currentDate, _destStorageAccountName, _destStorageKeyValue));
+            deleteTasks.Add(TryDeleteContainerAsync(_catalogStorageContainer, currentDate, _destStorageAccountName, _destStorageKeyValue));
+            deleteTasks.Add(TryDeleteContainerAsync(_registrationStorageContainer, currentDate, _destStorageAccountName, _destStorageKeyValue));
+            deleteTasks.Add(TryDeleteContainerAsync(_registrationgzStorageContainer, currentDate, _destStorageAccountName, _destStorageKeyValue));
+
+            await TryCopyContainerAsync(_luceneStorageContainer, currentDate, _luceneStorageAccountName, _luceneStorageKeyValue);
+            await TryCopyContainerAsync(_catalogStorageContainer, currentDate, _catalogStorageAccountName, _catalogStorageKeyValue);
+            await TryCopyContainerAsync(_registrationStorageContainer, currentDate, _registrationStorageAccountName, _registrationStorageKeyValue);
+            await TryCopyContainerAsync(_registrationgzStorageContainer, currentDate, _registrationStorageAccountName, _registrationStorageKeyValue);
```

Comment:
We need a staged copy. We decided on a specific order. 
PR: https://github.com/NuGet/NuGet.Jobs/pull/151